src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -147,94 +132,77 @@ public NetworkPeerServer(Network network, ProtocolVersion version, int internalP
         /// Starts listening on the server's initialialized endpoint.
         /// </summary>
         /// <param name="maxIncoming">Maximal number of newly connected clients waiting to be accepted.</param>
-        public void Listen(int maxIncoming = 8)
+        public void Listen(int maxIncoming = 20)
         {
             this.logger.LogTrace("({0}:{1})", nameof(maxIncoming), maxIncoming);
 
-            if (this.socket != null)
-            {
-                this.logger.LogTrace("(-)[ALREADY_LISTENING]");
-                throw new InvalidOperationException("Already listening");
-            }
-
             try
             {
-                this.socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
-                this.socket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.IPv6Only, false);
-
-                this.socket.Bind(this.LocalEndpoint);
-                this.socket.Listen(maxIncoming);
-
-                this.logger.LogTrace("Listening started.");
-                this.BeginAccept();
+                this.tcpListener.Server.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, true);
+                this.tcpListener.Start();
+                this.AcceptClientsAsync();
             }
-            catch (Exception ex)
+            catch (Exception e)
             {
-                this.logger.LogTrace("Exception occurred: {0}", ex.ToString());
+                this.logger.LogTrace("Exception occurred: {0}", e.ToString());
                 throw;
             }
 
             this.logger.LogTrace("(-)");
         }
 
         /// <summary>
-        /// Starts accepting connections from newly connected clients.
+        /// Implements loop accepting connections from newly connected clients.
```

Comment:
you're accepting connection from newly connected clients or from new clients wishing to connect?

Reply:
you are right, it is confusing 

will change to `Implements loop accepting new connections.`